ntbs-service/Services/SpecimenService.cs

```diff
@@ -19,57 +29,115 @@ public class SpecimenService : ISpecimenService
+            int notificationId)
         {
             using (var connection = new SqlConnection(_reportingDbConnectionString))
             {
                 connection.Open();
-                return await connection.QueryAsync<Specimen>(getMatchedSpecimenSqlFunction, new { notificationId });
+                return await connection.QueryAsync<MatchedSpecimen>(
+                    SpecimenQueryHelper.GetMatchedSpecimensForNotificationQuery,
+                    new {param = notificationId});
             }
         }
 
+        public async Task<IEnumerable<UnmatchedSpecimen>> GetAllUnmatchedSpecimensAsync()
+        {
+            var query = SpecimenQueryHelper.GetAllUnmatchedSpecimensQuery;
+            var unmatchedQueryResultRows = await ExecuteUnmatchedSpecimenQuery(query);
+            return GroupUnmatchedSpecimenRowsByReferenceNumber(unmatchedQueryResultRows);
+        }
+
+        public async Task<IEnumerable<UnmatchedSpecimen>> GetUnmatchedSpecimensDetailsForTbServicesAsync(
+            IEnumerable<string> tbServiceCodes)
+        {
+            var query = SpecimenQueryHelper.GetUnmatchedSpecimensForTbServicesQuery;
+            var formattedTbServiceCodes = SpecimenQueryHelper.FormatEnumerableParams(tbServiceCodes);
+            var unmatchedQueryResultRows = await ExecuteUnmatchedSpecimenQuery(query, formattedTbServiceCodes);
+            return GroupUnmatchedSpecimenRowsByReferenceNumber(unmatchedQueryResultRows);
+        }
+
+        public async Task<IEnumerable<UnmatchedSpecimen>> GetUnmatchedSpecimensDetailsForPhecsAsync(
+            IEnumerable<string> phecCodes)
+        {
+            var query = SpecimenQueryHelper.GetUnmatchedSpecimensForPhecsQuery;
+            var formattedPhecCodes = SpecimenQueryHelper.FormatEnumerableParams(phecCodes);
+            var unmatchedQueryResultRows = await ExecuteUnmatchedSpecimenQuery(query, formattedPhecCodes);
+            return GroupUnmatchedSpecimenRowsByReferenceNumber(unmatchedQueryResultRows);
+        }
+
+        private async Task<IEnumerable<UnmatchedQueryResultRow>> ExecuteUnmatchedSpecimenQuery(
+            string query,
+            string param = null)
+        {
+            using (var connection = new SqlConnection(_reportingDbConnectionString))
+            {
+                connection.Open();
+                return await connection.QueryAsync<SpecimenBase, SpecimenPotentialMatch, UnmatchedQueryResultRow>(
+                    query,
+                    (specimen, potentialMatch) => new UnmatchedQueryResultRow
+                    {
+                        SpecimenBase = specimen, SpecimenPotentialMatch = potentialMatch
+                    },
+                    splitOn: nameof(SpecimenPotentialMatch.NotificationId),
+                    param: new {param});
+            }
+        }
+
+        private static IEnumerable<UnmatchedSpecimen> GroupUnmatchedSpecimenRowsByReferenceNumber(

```

Comment:
I'd rename away from Group to Get or similar as the return of this is more just the list of specimens.

After chatting about this we are grouping at the bottom of this function - possibly needs a new line to highlight it or I might lean towards GroupPotientialMatchesByUnmatchedSpecimens (aka directed more at the return type from the grouping rather than the grouping of the rows going in). Happy to be overruled here!